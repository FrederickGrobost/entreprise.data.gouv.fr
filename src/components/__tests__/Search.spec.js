import Search from "@/components/Search.vue";
import { createLocalVue, shallowMount, mount } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);
jest.mock("@/store/index.js");
jest.mock("@/router/index.js");

describe("Results.vue", () => {
  let storeMocks;
  let wrapperSearch;
  let $route;

  beforeEach(() => {
    $route = {
      params: "mock-params",
      push: jest.fn(),
      path: "mock-path",
      query: {
        page: 2,
        fullText: "mock-fullText"
      }
    };

    storeMocks = createStoreMocks();

    wrapperSearch = shallowMount(Search, {
      localVue,
      store: storeMocks.store,
      mocks: { $route },
      stubs: ["router-link", "router-view"]
    });
  });

  test("Computed value showBackToResultsButton is falsy if not on etablissement page, \
      if numberResults > 1, and status === 200", () => {
    $route = {
      params: "mock-params",
      push: jest.fn(),
      path: "/results/mock-params",
      query: {
        page: 2,
        fullText: "mock-fullText"
      }
    };
    wrapperSearch = shallowMount(Search, {
      localVue,
      store: new Vuex.Store({
        state: {
          welcomeText: {
            isWelcomeTextVisible: true
          },
          results: {
            status: 200
          },
          search: {
            storedFullText: "mock-storedFullText"
          }
        },
        mutations: {
          setPage: jest.fn(),
          setFullText: jest.fn()
        },
        actions: {
          requestSearchFullText: jest.fn()
        },
        getters: {
          numberResults: jest.fn().mockReturnValue(2)
        }
      }),
      mocks: { $route },
      stubs: ["router-link", "router-view"]
    });
    expect(wrapperSearch.vm.showBackToResultsButton).toBeFalsy();
  });

  test("Computed value showBackToResultsButton is falsy if on etablissement page, \
    if numberResults === 1, and status === 200", () => {
    $route = {
      params: "mock-params",
      push: jest.fn(),
      path: "/results/mock-params",
      query: {
        page: 2,
        fullText: "mock-fullText"
      }
    };
    wrapperSearch = shallowMount(Search, {
      localVue,
      store: new Vuex.Store({
        state: {
          welcomeText: {
            isWelcomeTextVisible: true
          },
          results: {
            status: 200
          },
          search: {
            storedFullText: "mock-storedFullText"
          }
        },
        mutations: {
          setPage: jest.fn(),
          setFullText: jest.fn()
        },
        actions: {
          requestSearchFullText: jest.fn()
        },
        getters: {
          numberResults: jest.fn().mockReturnValue(2)
        }
      }),
      mocks: { $route },
      stubs: ["router-link", "router-view"]
    });
    expect(wrapperSearch.vm.showBackToResultsButton).toBeFalsy();
  });
  test("Computed value showBackToResultsButton is falsy only on etablissement page, \
    if numberResults > 1, and status !== 200", () => {
    $route = {
      params: "mock-params",
      push: jest.fn(),
      path: "/etablissement/mock-searchId",
      query: {
        page: 2,
        fullText: "mock-fullText"
      }
    };
    wrapperSearch = shallowMount(Search, {
      localVue,
      store: new Vuex.Store({
        state: {
          welcomeText: {
            isWelcomeTextVisible: true
          },
          results: {
            status: 404
          },
          search: {
            storedFullText: "mock-storedFullText"
          }
        },
        mutations: {
          setPage: jest.fn(),
          setFullText: jest.fn()
        },
        actions: {
          requestSearchFullText: jest.fn()
        },
        getters: {
          numberResults: jest.fn().mockReturnValue(2)
        }
      }),
      mocks: { $route },
      stubs: ["router-link", "router-view"]
    });
    expect(wrapperSearch.vm.showBackToResultsButton).toBeFalsy();
  });

  test("At creation, if there is a query page, we set the correct page", () => {
    expect(storeMocks.mutations.setPage).toHaveBeenCalledWith(
      storeMocks.state,
      2
    );
  });

  // TODO: test when router-link will be testable
  // test('Clicking back to results buttons goes back to result page')
});

describe("Search.vue snapshot", () => {
  const storeMocks = createStoreMocks();

  const $route = {
    params: "mock-params",
    push: jest.fn(),
    path: "mock-path",
    query: {
      page: 2,
      fullText: "mock-fullText"
    }
  };

  const wrapperSearch = mount(Search, {
    localVue,
    store: storeMocks.store,
    mocks: { $route },
    stubs: ["router-link", "router-view"]
  });

  test("It match the snapshot when there arent results", () => {
    expect(wrapperSearch.vm.$el).toMatchSnapshot();
  });

  // TODO: test when router-link will be testable
  // test('It match the snapshot when there are results')
});
