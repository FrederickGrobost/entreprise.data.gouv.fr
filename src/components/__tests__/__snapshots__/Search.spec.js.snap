// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue snapshot It match the snapshot when there arent results 1`] = `
<div
  class="hero"
  role="banner"
>
  <div
    class="hero__container container"
  >
    <div
      class="text-center"
      name="fade"
    >
      <h1
        class="search__title"
      >
        
          Retrouvez toutes les informations publiques concernant les entreprises et associations de France
        
      </h1>
       
      <p
        class="search__subtitle"
      >
        Les bases de données sur l’état civil des entreprises et associations françaises sont maintenant accessibles à tous, sans frais.
      </p>
    </div>
     
    <div
      class="form__group"
    >
      <input
        name="search"
        placeholder="Nom, SIREN, SIRET, adresse..."
        type="text"
      />
       
      <button
        class="overlay-button"
      >
        <svg
          class="icon icon-search"
        >
          <use
            xlink:href="#icon-search"
          />
        </svg>
      </button>
       
      <ul
        class="suggestions"
        style="display: none;"
      />
    </div>
     
    <!---->
  </div>
</div>
`;
